### PR TITLE
rgw/sts: adding error log messages to STS code.

### PR DESCRIPTION
**Pull request by pritha-srivastava**
Signed-off-by: Pritha Srivastava <prsrivas@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by pritha-srivastava** [comment]
Teuthology run: https://pulpito.ceph.com/prsrivas-2020-07-15_17:03:41-rgw-wip-rgw-pritha-test-fixes-distro-basic-smithi/

**Comment by pritha-srivastava** [comment]
> Teuthology run: https://pulpito.ceph.com/prsrivas-2020-07-15_17:03:41-rgw-wip-rgw-pritha-test-fixes-distro-basic-smithi/

The failures are:

1. Command failed on smithi045 with status 1: "cd /home/ubuntu/cephtest/keystone && source .tox/venv/bin/activate && openstack service create --os-username admin --os-password ADMIN --os-user-domain-id default --os-project-name admin --os-project-domain-id default --os-identity-api-version 3 --os-auth-url http://smithi045.front.sepia.ceph.com:35357/v3 --description 'Swift Service' --name swift object-store --debug"

2. rgw multisite, test failures

The failures aren't related to changes in this PR, so this is ready to be merged.

**Comment by pritha-srivastava** [comment]
jenkins retest this please

**Comment by pritha-srivastava** [comment]
jenkins retest this please

**Comment by mattbenjamin** [comment]
jenkins retest this please

**Comment by mattbenjamin** [comment]
@pritha-srivastava if this hasn't been rebased in the last week, could you please do so?  I think this may be needed to address ceph dashboard failures.  procedure should be to start a new fork of ceph master and then "git cherry pick" each of the changes here in order to the new branch;  doing so won't require you to do another teuthology run

**Comment by pritha-srivastava** [comment]
@mattbenjamin : I have rebased this PR and all checks have passed and the teuthology run is also successful, hence it is ready to be merged.

File: src/rgw/rgw_rest_sts.cc [hunk 1]
```diff
@@ -362,14 +367,16 @@ int RGWREST_STS::verify_permission()
     // If yes, then return 0, else -EPERM
     auto p_res = p.eval_principal(s->env, *s->auth.identity);
     if (p_res == rgw::IAM::Effect::Deny) {
+      ldout(s->cct, 0) << "evaluating principal returned deny" << dendl;
```
Comment (mattbenjamin) [inline src/rgw/rgw_rest_sts.cc:370]: am I correct that this just means that "authz failed" for some request? in this one message at leaset, we aren't adding any detail to the log, other than "x failed"--is that sufficient?  also, this is at level 0, which implies that this is "basic info" on what happened--does other logging already say "attempting x as user y failed"?  if so, maybe this one should be at a higher debug level?

File: src/rgw/rgw_rest_sts.cc [hunk 2]
```diff
@@ -524,7 +534,7 @@ int RGWSTSAssumeRole::get_params()
       const rgw::IAM::Policy p(s->cct, s->user->get_tenant(), bl);
     }
     catch (rgw::IAM::PolicyParseException& e) {
-      ldout(s->cct, 20) << "failed to parse policy: " << e.what() << "policy" << policy << dendl;
+      ldout(s->cct, 0) << "failed to parse policy: " << e.what() << "policy" << policy << dendl;
```
Comment (mattbenjamin) [inline src/rgw/rgw_rest_sts.cc:537]: this looks ok

File: src/rgw/rgw_sts.cc [hunk 3]
```diff
@@ -68,18 +68,20 @@ int Credentials::generateCredentials(CephContext* cct,
...
   if (! cryptohandler) {
+    ldout(cct, 0) << "ERROR: No AES cryto handler found !" << dendl;
     return -EINVAL;
   }
   string secret_s = cct->_conf->rgw_sts_key;
   buffer::ptr secret(secret_s.c_str(), secret_s.length());
   int ret = 0;
   if (ret = cryptohandler->validate_secret(secret); ret < 0) {
-    ldout(cct, 0) << "ERROR: Invalid secret key" << dendl;
+    ldout(cct, 0) << "ERROR: Invalid rgw sts key, please ensure its length is 16" << dendl;
```
Comment (mattbenjamin) [inline src/rgw/rgw_sts.cc:78]: good message

**Review by mattbenjamin — APPROVED** [review]
I was a bit worried a couple of messages are verbose but not adding info, but overall, this detail seems critical to help sites debug problems with idp integration

**Pull request merged**